keygen/src/keygen.rs

```diff
@@ -350,101 +348,104 @@ fn main() -> Result<(), Box<dyn error::Error>> {
                 )
                 .arg(
-                    Arg::with_name("keypair")
+                    Arg::new("keypair")
                         .index(2)
                         .value_name("KEYPAIR")
                         .takes_value(true)
                         .help("Filepath or URL to a keypair"),
                 )
         )
         .subcommand(
-            SubCommand::with_name("new")
+            Command::new("new")
                 .about("Generate new keypair file from a random seed phrase and optional BIP39 passphrase")
-                .setting(AppSettings::DisableVersion)
+                .disable_version_flag(true)
                 .arg(
-                    Arg::with_name("outfile")
-                        .short("o")
+                    Arg::new("outfile")
+                        .short('o')
                         .long("outfile")
                         .value_name("FILEPATH")
                         .takes_value(true)
                         .help("Path to generated file"),
                 )
                 .arg(
-                    Arg::with_name("force")
-                        .short("f")
+                    Arg::new("force")
+                        .short('f')
                         .long("force")
                         .help("Overwrite the output file if it exists"),
                 )
                 .arg(
-                    Arg::with_name("silent")
-                        .short("s")
+                    Arg::new("silent")
+                        .short('s')
                         .long("silent")
                         .help("Do not display seed phrase. Useful when piping output to other programs that prompt for user input, like gpg"),
                 )
                 .key_generation_common_args()
                 .arg(no_outfile_arg())
         )
         .subcommand(
-            SubCommand::with_name("grind")
+            Command::new("grind")
                 .about("Grind for vanity keypairs")
-                .setting(AppSettings::DisableVersion)
+                .disable_version_flag(true)
                 .arg(
-                    Arg::with_name("ignore_case")
+                    Arg::new("ignore_case")
                         .long("ignore-case")
                         .help("Performs case insensitive matches"),
                 )
                 .arg(
-                    Arg::with_name("starts_with")
+                    Arg::new("starts_with")
                         .long("starts-with")
                         .value_name("PREFIX:COUNT")
                         .number_of_values(1)
                         .takes_value(true)
-                        .multiple(true)
-                        .validator(grind_validator_starts_with)
+                        .multiple_occurrences(true)
+                        .multiple_values(true)
+                        .validator(|s| grind_validator_starts_with(s.to_string()))
```

Comment:
doesn't look like `grind_validator_starts_with` strictly requires a `String`. I'd prefer to change its signature to take `&str`

Reply:
thanks for pointing out, fixed all the occurrences except for one which is more elaborated